src/palm-react-native-ui-kit/screens/channels/CreateChannelScreen/TokenGating.tsx

```diff
@@ -56,152 +60,204 @@ const TokenGating = ({
     limit: 30,
   })
 
+  const userBalances = useUserBalance({
+    address: user?.address!,
+    chain: gatingTokenNetwork,
+  })
+  const userNfts = useUserNftList({
+    userAddress: user?.address!,
+    selectedNetwork: gatingTokenNetwork,
+  })
+
+  const [enableDoneButton, setEnableDoneButton] = useState(false)
+
+  useEffect(() => {
+    if (step !== 2 || userBalances.isLoading || userNfts.loading) {
+      return
+    }
+
+    if (userNfts.hasNextPage) {
+      userNfts.fetchNextPage()
+      return
+    }
+
+    if (selectedGatingToken.gatingType === 'NFT') {
+      // NFT
+      const count = userNfts.items.filter(
+        i => i.name === selectedGatingToken.name
+      ).length
+      const input = Number(gatingTokenAmount)
+
+      setEnableDoneButton(input > 0 && count >= input)
+    } else {
+      // Native
+      const balance = UTIL.toBn(userBalances.balance)
+      const input = UTIL.toBn(UTIL.microfyP(gatingTokenAmount as Token))
+
+      setEnableDoneButton(!input.isZero() && balance.gte(input))
+    }
+  }, [gatingTokenAmount, userBalances, userNfts])
```

Comment:
why is this needed? Also, for performance, can we make this to query userBalances or userNfts depending on the token gating type?

Reply:
https://klaytn-foundation.atlassian.net/browse/WEB3-198

The view is not separated, and token selection and value input are being done at the same view. 
I will refactor this.